https://github.com/fail2ban/fail2ban/pull/3037
https://bugs.gentoo.org/794931

From: Mike Gilbert <floppym@gentoo.org>
Date: Tue, 8 Jun 2021 17:48:12 -0400
Subject: [PATCH] tests: improve detection of readable systemd journal

Look for system.journal in journal sub-directory.
Add -readable to the find command.

Bug: https://bugs.gentoo.org/794931
--- a/fail2ban/tests/filtertestcase.py
+++ b/fail2ban/tests/filtertestcase.py
@@ -1396,7 +1396,7 @@ def _getRuntimeJournal(self):
 				# check one at at time until the first hit
 				for systemd_var in 'system-runtime-logs', 'system-state-logs':
 					tmp = Utils.executeCmd(
-						'find "$(systemd-path %s)" -name system.journal' % systemd_var,
+						'find "$(systemd-path %s)/journal" -name system.journal -readable' % systemd_var,
 						timeout=10, shell=True, output=True
 					)
 					self.assertTrue(tmp)
